synapse/lib/logging.py

```diff
@@ -0,0 +1,211 @@
+
+        loginfo = {
+            'message': record.message,
+            'logger': {
+                'name': record.name,
+                'func': record.funcName,
+            },
+            'level': record.levelname,
+            'time': self.formatTime(record, self.datefmt),
+        }
+
+        loginfo.update(_glob_loginfo)
+
+        if hasattr(record, 'loginfo'):
+            loginfo.update(record.loginfo)
+
+        if (user := s_scope.get('user')) is not None:
+            loginfo['user'] = user.iden
+            loginfo['username'] = user.name
+
+        elif (sess := s_scope.get('sess')) is not None:
+            if sess.user is not None:
+                loginfo['user'] = sess.user.iden
+                loginfo['username'] = sess.user.name
+
+        if record.exc_info:
+            loginfo['error'] = s_common.excinfo(record.exc_info[1])
+
+        if not hasattr(record, 'params'):
+            record.params = {}
+
+        loginfo['params'] = record.params
+
+        _addLogInfo(loginfo)
+
+        return loginfo
+
+    def format(self, record):
+        loginfo = self.genLogInfo(record)
+        return json.dumps(loginfo, default=str)
+
+class TextFormatter(Formatter):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['fmt'] = s_const.LOG_FORMAT
+        return super().__init__(*args, **kwargs)
+
+    def format(self, record):
+        loginfo = self.genLogInfo(record)
+        return logging.Formatter.format(self, record)
+
+_glob_logconf = {}
+def setup(**conf):
+    '''
+    Configure synapse logging.
+    '''
+    conf.update(getLogConfFromEnv())
+
+    if conf.get('level') is None:
+        conf['level'] = logging.WARNING
+
+    if conf.get('structlog') is None:
+        conf['structlog'] = False
+
+    fmtclass = Formatter
+    if not conf.get('structlog'):
+        fmtclass = TextFormatter
+
+    # this is used to pass things like service name
+    # to child processes and forked workers...
+    loginfo = conf.pop('loginfo', None)
+    if loginfo is not None:
+        _glob_loginfo.update(loginfo)
+
+    handler = logging.StreamHandler()
+    handler.setFormatter(fmtclass(datefmt=conf.get('datefmt')))
+
+    level = normLogLevel(conf.get('level'))
+
+    _glob_logconf.clear()
+    _glob_logconf.update(conf)
+
+    logging.basicConfig(level=level, handlers=(handler,))
+
+    logger.info('log level set to %s', s_const.LOG_LEVEL_INVERSE_CHOICES.get(level))
+
+    return conf
+
+def getLogConf():
+    logconf = _glob_logconf.copy()
+    logconf['loginfo'] = _glob_loginfo.copy()
+    return logconf
+
+def getLogConfFromEnv():
+
+    conf = {}
+
+    if (level := os.getenv('SYN_LOG_LEVEL')) is not None:
+        conf['level'] = normLogLevel(level)
+
+    if (datefmt := os.getenv('SYN_LOG_DATEFORMAT')) is not None:
+        conf['datefmt'] = datefmt
+
+    if (structlog := os.getenv('SYN_LOG_STRUCT')) is not None:
+        conf['structlog'] = structlog.lower() in ('1', 'true')
```

Comment:
```suggestion
    conf['structlog'] = s_common.envbool('SYN_LOG_STRUCT')
```

Reply:
This is incorrect. If the env var is not set, we should not populate the log conf. The only way to use `envbool` would end up calling `os.getenv()` twice. Not opposed to the idea, but worth a quick goback.

Reply:
What's the difference between `structlog` not being present and being set to False? Maybe this instead?

```suggestion
    if s_common.envbool('SYN_LOG_STRUCT'):
        conf['structlog'] = True
```